src/window.ts

```diff
@@ -262,6 +272,46 @@ export class ShellWindow {
             return xprop.get_xid(this.meta);
         })
     }
+
+    show_border() {
+        if (this.ext.settings.active_hint()) {
+            let border = this.border;
+            if (!this.is_maximized()) {
+                border.show();
+            }
+        }
+    }
+
+    hide_border() {
+        let border = this.border;
+        border.hide();
+    }
+
+    update_border_layout() {
+        let frameRect = this.meta.get_frame_rect();
+        let [frameX, frameY, frameWidth, frameHeight] = [frameRect.x, frameRect.y, frameRect.width, frameRect.height];
+
+        let border = this.border;
+
+        border.set_position(frameX, frameY);
+        border.set_size(frameWidth, frameHeight);
+    }
+
+    bind_window_events() {
+        // @ts-ignore
+        let windowSignals = [
+            this.meta.connect('size-changed', () => {
+                this.ext.show_border_on_focused();
+                if (this.is_maximized() || this.meta.minimized) {
+                    this.hide_border();
+                }
+                this.update_border_layout();
+            }),
+            this.meta.connect('position-changed', () => {
+                this.update_border_layout();
+            }),
+        ];
+    }
```

Comment:
Would this work here? https://github.com/pop-os/shell/blob/master_focal/src/extension.ts#L381:L386

Reply:
It does not seem to work when I bounded the handler. That's why I added two new sets.

Reply:
Then we should definitely add these signals somewhere so we can disconnect them on extension disabling / window destruction.

Reply:
@mmstick - added on `this.ext.window_signals` push on location of binding